packages/txservice/src/monitor/monitor.ts

```diff
@@ -0,0 +1,242 @@
+      throw result.error;
+    }
+    // If transaction buffer above returns null, that indicates the buffer is empty; meaning
+    // we haven't sent any transactions yet, and we'll use this value here.
+    const pending = result.value;
+    // Increment if we got back a value from buffer as we want the value of last tx's nonce + 1.
+    return Math.max(buffer + 1, pending);
+  }
+
+  private async backfillLoop() {
+    // TODO: Make sure this loop is throw-proof
+    while (true) {
+      delay(1000);
+      const pending = this.buffer.pending;
+      // Lazy solution: we only care about a potential hold-up if it could hold anything up.
+      if (pending.length < 2) {
+        continue;
+      }
+      const result = await this.provider.getTransactionCount();
+      if (result.isErr()) {
+        // TODO: log
+      }
+      const currentNonce = result.isOk() ? result.value : -1;
+      const tx: Transaction | undefined = this.buffer.get(currentNonce);
+      if (tx == null) {
+        // BC it could just be the latest nonce and we're currently not doing anything
+        // TODO: backfill
+        this.backfill(currentNonce, undefined, "NOT_FOUND");
+      } else {
+        if (tx.didFinish || tx.isBackfill) {
+          // IF the transaction did finish already, or this is already being backfilled (from a previous iteration
+          // here), we can ignore this one.
+          // TODO: Do we actually want to proceed even if a tx IS a backfill? What would that accomplish? Because if
+          // even the backfill isn't working, then how would ANOTHER backfill solve anything? In fact, maybe we want to
+          // shut things down if even a backfill tx isn't going through?
+          continue;
+        }
+        // Check to make sure that the transaction has leftover time to live.
+        const ttl = tx.timeUntilExpiry();
+        if (ttl < 0) {
+          this.backfill(currentNonce, tx, "EXPIRED");
+        } else {
+          if (tx.attempt > 5) {
+            try {
+              // You get 1 hail mary. (This should throw if it fails).
+              await this.hailMary(tx);
+            } catch (error) {
+              // If it fails from there, kill tx and backfill
+              await tx.kill(error);
+              this.backfill(currentNonce, tx, "TTL_HM_FAILED", { error });
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private async backfill(nonce: number, blockade: Transaction | undefined, reason: string, context: any = {}) {
+    const method = this.backfill.name;
+
+    const addressResult = await this.provider.getAddress();
+    if (addressResult.isErr()) {
+      throw addressResult.error;
+    }
+    // Sending a 0 wei transaction to fill the gap. We only have to eat the cost of gas here.
+    const minTx: WriteTransaction = {
+      chainId: this.chainId,
+      value: BigNumber.from(0),
+      data: "0x",
+      to: addressResult.value,
+    };
+    const gas = await this.getGas(minTx);
```

Comment:
we should not be resubmitting this backfill tx multiple times. instead we should minimize the number of times were "backfilling" but expedite that as much as possible by sending it at a high gas. we definitely dont want to end up in situations where we are backfilling transactions, it gets added to the buffer, and then we are waiting for the gas to get bumped sufficiently high so it eventually goes through

open to counterarguments though, since obviously the "f it send it at max" isn't an elegant solution

Reply:
No, it's the solution we definitely want for now. but leaving this comment thread open in case @rhlsthrm has opinions